libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -651,16 +652,19 @@ string IRGenerator::dispatchRoutine(ContractDefinition const& _contract)
 	return t.render();
 }
 
-string IRGenerator::memoryInit()
+string IRGenerator::memoryInit(bool _useMemoryGuard)
 {
 	// This function should be called at the beginning of the EVM call frame
 	// and thus can assume all memory to be zero, including the contents of
 	// the "zero memory area" (the position CompilerUtils::zeroPointer points to).
-	return
-		Whiskers{"mstore(<memPtr>, <freeMemoryStart>)"}
-		("memPtr", to_string(CompilerUtils::freeMemoryPointer))
-		("freeMemoryStart", to_string(CompilerUtils::generalPurposeMemoryStart + m_context.reservedMemory()))
-		.render();
+	return Whiskers{
+			_useMemoryGuard ?
+			"mstore(<memPtr>, memoryguard(<freeMemoryStart>))" :
+			"mstore(<memPtr>, <freeMemoryStart>)"
+		}("memPtr", to_string(CompilerUtils::freeMemoryPointer))(
```

Comment:
Nit: the indentation previously was nicer:
```
		("memPtr", to_string(CompilerUtils::freeMemoryPointer))
		("freeMemoryStart", to_string(CompilerUtils::generalPurposeMemoryStart + m_context.reservedMemory()))
		.render();
```

We need clang-tidy :)

Reply:
This is in accordance with the coding style actually :-).

Reply:
Still changed it slightly.